pages/members/index.tsx

```diff
@@ -1,29 +1,49 @@
 import { observer } from 'mobx-react';
-import { PropsWithoutRef, PureComponent } from 'react';
+import { PureComponent } from 'react';
 import { Container } from 'react-bootstrap';
 import { Loading } from 'idea-react';
+import { observable } from 'mobx';
+import { isFunction } from 'lodash';
 
 import PageHead from '../../components/PageHead';
 import membersStore, { Member } from '../../models/Members';
 import { LazyLoad } from '../../utils/layzload';
-import MembersTitle from '../../components/Members/MembersTitle';
-import MembersList from '../../components/Members/MembersList';
-import MembersTabs, {
+import { MembersTitle } from '../../components/Members/MembersTitle';
+import { MembersList } from '../../components/Members/MembersList';
+import {
+  MembersTabs,
   MembersTabsProps,
 } from '../../components/Members/MembersTabs';
 
-type MembersGroup = PropsWithoutRef<{
-  [proppName: string]: MembersTabsProps;
-}>;
+type MembersGroup = Record<
+  string,
+  {
+    [groupName: string]: MembersTabsProps;
+  }
+>;
+
+const groupMemberFn = (
+  groupMap: MembersGroup,
+  groupKeys: string[],
+  groupItem: Member,
+  insertFn?: Function,
+) => {
+  if (!groupKeys?.length) return;
+  isFunction(insertFn) && insertFn();
+  groupKeys.forEach((key: string) =>
+    ((groupMap[key] = groupMap[key] || { list: [] }).list as Member[]).push(
+      groupItem,
+    ),
+  );
```

Comment:
```suggestion
) => {
  for (const key of groupKeys)
    ((groupMap[key] ||= { list: [] }).list as Member[]).push(groupItem);
```
参数类型声明 `groupKeys` 一定存在，空数组会自动跳过迭代，故在最前的回调函数就可以放到外面执行。

Reply:
好的 我放到最外面 也行 groupKeys 根据目前数据不规范 或者未分组撒的可能没有从😂